sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TablesJacksonSerializer.java

```diff
@@ -94,9 +96,13 @@ public <U> U deserialize(String value, Type type, SerializerEncoding serializerE
     public <U> U deserialize(InputStream inputStream, Type type, SerializerEncoding serializerEncoding)
         throws IOException {
         if (inputStream != null && type == TableEntityQueryResponse.class) {
-            return deserializeTableEntityQueryResponse(super.serializer().readTree(inputStream));
+            try (JsonReader jsonReader = JsonProviders.createReader(inputStream)) {
```

Comment:
only try?

Reply:
I don't think I understand this comment, is this asking why the `InputStream` isn't in the try? If so, it wasn't before and I think that can be looked at in a follow-up to see if it needs to be closed as well.

Reply:
Sorry what I am asking is If we only have `try(....)` without catch the exception or add a finally block?

Reply:
Ah, this is a try-with-resource block. `JsonReader` implements `Closeable` and this will close it once it's done being used so you don't need to write

```java
JsonReader jsonReader = JsonProviders.createReader(inputStream);
try {
    return deserializeTableEntityQueryResponse(jsonReader);
} finally {
    jsonReader.close();
}
```